packages/state-transition/src/util/deposit.ts

```diff
@@ -0,0 +1,21 @@
+import {ForkSeq, MAX_DEPOSITS} from "@lodestar/params";
+import {UintNum64} from "@lodestar/types";
+import {CachedBeaconStateAllForks, CachedBeaconStateEIP6110} from "../types.js";
+
+export function getEth1DepositCount(state: CachedBeaconStateAllForks): UintNum64 {
+  if (state.config.getForkSeq(state.slot) >= ForkSeq.eip6110) {
+    const eip6110State = state as CachedBeaconStateEIP6110;
+    const eth1DataIndexLimit =
+      state.eth1Data.depositCount < eip6110State.depositReceiptsStartIndex
+        ? state.eth1Data.depositCount
+        : eip6110State.depositReceiptsStartIndex;
```

Comment:
What if `eip6110State.depositReceiptsStartIndex` is the max value? I would prefer for the BigNumber to Number casting to happen here, properly guarded by upper bounds, instead of below. Also use
```
Number(x as bigint)
```
instead of
```
Number(x)
```
or a similar trick since Number constructor types suck and accept `any`

Reply:
> What if `eip6110State.depositReceiptsStartIndex` is the max value? 

Can you elaborate? I believe we can safely compare `UintNum64` and `UintBn64`. 
Essentially this statement is `Math.min(state.eth1Data.depositCount, eip6110State.depositReceiptsStartIndex)` so `Math.min(UintNum64, UintBn64)`